projects/saturn/src/main/java/io/fairspace/saturn/services/metadata/MetadataService.java

```diff
@@ -190,17 +199,59 @@ public void patch(Model model) {
 
     private Set<Resource> update(Model modelToRemove, Model modelToAdd) {
         return transactions.calculateWrite(before -> {
-            trimLabels(modelToAdd);
-            var after = updatedView(before, modelToRemove, modelToAdd);
+            Set<Resource> updatedResources = new HashSet<>();
 
-            validate(before, after, modelToRemove, modelToAdd);
+            updatedResources.addAll(updateLabelsAndLinkedDavResources(before, modelToAdd));
 
+            var after = updatedView(before, modelToRemove, modelToAdd);
+            validate(before, after, modelToRemove, modelToAdd);
             persist(modelToRemove, modelToAdd);
+            updatedResources.addAll(modelToRemove.listSubjects().andThen(modelToAdd.listSubjects()).toSet());
 
-            return modelToRemove.listSubjects().andThen(modelToAdd.listSubjects()).toSet();
+            return updatedResources;
         });
     }
 
+    private Set<Resource> updateLabelsAndLinkedDavResources(Model before, Model toAdd) {
+        Set<Resource> updatedDavResources = new HashSet<>();
+        toAdd.listSubjectsWithProperty(RDFS.label)
```

Comment:
does 'toAdd' only contain linked resources? otherwise everything with this label gets updated which should not be done.

Reply:
`toAdd` contains all the labels that were changed in a single request. It needs to take all of them to trim them. However, before renaming, it filters on directory types that are linked to a resource (see line 230)